core/app/models/spree/app_configuration.rb

```diff
@@ -136,6 +136,11 @@ class AppConfiguration < Preferences::Configuration
     #     charged (default: +14+)
     preference :expedited_exchanges_days_window, :integer, default: 14
 
+    # @!attribute [rw] generate_api_key_for_all_roles
+    #   @return [Boolean] Allow generating api key automatically for user
+    #   at role_user creation for all roles. (default: +false+)
+    preference :generate_api_key_for_all_roles, :boolean, default: false
```

Comment:
I know I (sort of) asked for this, but seeing the implementation I don't think this makes sense. Most users signing up to a store will have no role at all. This option will grant an api key to any user with any role (but they must have a role), which is probably not as useful and I think the distinction will be confusing.

I would still (eventually) like to see something for granting all users an API key, but I'd be :+1: just removing this option